src/sql/src/plan/query.rs

```diff
@@ -3066,10 +3081,12 @@ fn plan_and(
     right: &Expr<Aug>,
 ) -> Result<CoercibleScalarExpr, PlanError> {
     let ecx = ecx.with_name("AND argument");
-    Ok(HirScalarExpr::CallBinary {
-        func: BinaryFunc::And,
-        expr1: Box::new(plan_expr(&ecx, left)?.type_as(&ecx, &ScalarType::Bool)?),
-        expr2: Box::new(plan_expr(&ecx, right)?.type_as(&ecx, &ScalarType::Bool)?),
+    Ok(HirScalarExpr::CallVariadic {
+        func: VariadicFunc::And,
+        exprs: vec![
+            plan_expr(&ecx, left)?.type_as(&ecx, &ScalarType::Bool)?,
+            plan_expr(&ecx, right)?.type_as(&ecx, &ScalarType::Bool)?,
+        ],
```

Comment:
opportunity to use the new `HirScalarExpr::variadic_and` function

Reply:
Done